Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -594,7 +592,7 @@ class BrowserViewController: UIViewController {
     }
 
     private func presentContextualHint() {
-        if shouldShowIntroScreen { return }
+        if IntroScreenManager(prefs: profile.prefs).shouldShowIntroScreen { return }
         present(contextHintVC, animated: true)
```

Comment:
Can this be injected? I find it a bit awkward to keep instantiating every time we need to use it.

Reply:
It will be with the new code changes 😬 In the meantime I would be ok to just instantiate it to keep it simple until we move towards the new code.